### ISSUE TITLE
`infinite-loop` and `never_loop` trigger on the same loop 😆

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#16134]
```rust
#![warn(clippy::infinite_loop, clippy::never_loop)]
fn main() {
    'outer2: loop {
        let _: i32 = loop { loop { break 'outer2 } };
    }
}
```

```
...
warning: this loop never actually loops
 --> a.rs:4:22
  |
4 |         let _: i32 = loop { loop { break 'outer2 } };
  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#never_loop

warning: infinite loop detected
 --> a.rs:4:22
  |
4 |         let _: i32 = loop { loop { break 'outer2 } };
  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#infinite_loop
note: the lint level is defined here
...
```
